### DISCUSSION TITLE
Session could not start on Jenkins when running Appium Scripts

### DISCUSSION THREAD

**Discussion by surajb0710** [appium/appium#17081] [Q&A]
DesiredCapabilities cap = new DesiredCapabilities();

	
		cap.setCapability("appium:deviceName", "Redmi Note 8");
		cap.setCapability("appium:udid", "14eb3774");
		cap.setCapability("appium:platformName", "Android");
		cap.setCapability("platformVersion", "11");
		cap.setCapability("appium:appPackage", "com.spotify.music");
		cap.setCapability("appium:appActivity", "com.spotify.music.MainActivity");
		
		/*
		
		cap.setCapability("waitForQuietness", false);
		cap.setCapability("waitForQuiescence", false);
		cap.setCapability("wdaEventloopIdleDelay", 7);
		cap.setCapability("eventLoopIdleDelaySec", 4);
		cap.setCapability("startIWDP", true);
		
		*/
		
		URL url = new URL("http://127.0.0.1:4723/wd/hub");

		driver = new AppiumDriver(url, cap);

This is running fine on eclipse, but when I am trying to run on jenkins, I am getting below error

Running TestSuite
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Configuring TestNG with: org.apache.maven.surefire.testng.conf.TestNG652Configurator@19bb089b
ERROR StatusLogger Log4j2 could not find a logging implementation. Please add log4j-core to the classpath. Using SimpleLogger to log to the console...
Tests run: 5, Failures: 1, Errors: 0, Skipped: 4, Time elapsed: 2.916 sec <<< FAILURE!
setup(com.automation.test.RegScreenTest)  Time elapsed: 2.366 sec  <<< FAILURE!
org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: An unknown server-side error occurred while processing the command. Original error: Neither ANDROID_HOME nor ANDROID_SDK_ROOT environment variable was exported. Read https://developer.android.com/studio/command-line/variables for more details

pls guide how to fix this. TIA.

**Comment by jlipps** [comment]
The answer is in the error message: `Neither ANDROID_HOME nor ANDROID_SDK_ROOT environment variable was exported.`. You need to make sure to set these env vars appropriately in your Jenkins setup I guess.

**Reply by surajb0710** [reply]
I added the SDK path to global settings, however still getting the same error.

**Reply by jlipps** [reply]
then somehow it's not getting set correctly. you could try to echo it out in your logs to double check that it is correct.
